runhouse/resources/hardware/cluster.py

```diff
@@ -418,49 +418,70 @@ def _sync_runhouse_to_cluster(self, _install_url=None, env=None):
 
         local_rh_package_path = Path(importlib.util.find_spec("runhouse").origin).parent
 
-        # Check if runhouse is installed from source and has setup.py
-        if (
+        installed_editable_locally = (
             not _install_url
             and local_rh_package_path.parent.name == "runhouse"
             and (local_rh_package_path.parent / "setup.py").exists()
-        ):
-            # Package is installed in editable mode
-            local_rh_package_path = local_rh_package_path.parent
-            dest_path = f"~/{local_rh_package_path.name}"
-
-            self._rsync(
-                source=str(local_rh_package_path),
-                dest=dest_path,
-                node="all",
-                up=True,
-                contents=True,
-                filter_options="- docs/",
-            )
-            rh_install_cmd = "python3 -m pip install ./runhouse"
-        # elif local_rh_package_path.parent.name == 'site-packages':
-        else:
-            # Package is installed in site-packages
-            # status_codes = self.run(['pip install runhouse-nightly==0.0.2.20221202'], stream_logs=True)
-            # rh_package = 'runhouse_nightly-0.0.1.dev20221202-py3-none-any.whl'
-            # rh_download_cmd = f'curl https://runhouse-package.s3.amazonaws.com/{rh_package} --output {rh_package}'
-            if not _install_url:
-                import runhouse
-
-                _install_url = f"runhouse=={runhouse.__version__}"
-            rh_install_cmd = f"python3 -m pip install {_install_url}"
-
-        for node in self.ips:
-            status_codes = self.run(
-                [rh_install_cmd],
-                node=node,
-                env=env,
-                stream_logs=True,
+        )
+
+        if installed_editable_locally or resync_rh:
+            remote_ray_version_call = self.run(
+                ["ray --version"], node="all", env=env, stream_logs=False
             )
+            ray_installed_remotely = remote_ray_version_call[0][0][0] == 0
+            if not ray_installed_remotely:
+                local_ray_version_call = subprocess.run(
+                    ["pip freeze | grep ray"],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    shell=True,
+                )
```

Comment:
Do you have an existing helper from the new package functionality to grab the local version for a package? Better not to have to handle edge cases here like editable / local installs.